src/frontend/src/webhook/mod.rs

```diff
@@ -129,45 +111,113 @@ pub(super) mod handlers {
+        builder.append(Some(jsonb_val.as_scalar_ref()));
+        let data_chunk = DataChunk::new(vec![builder.finish().into_ref()], 1);
+
+        // fill the data_chunk
+        fast_insert_request.data_chunk = Some(data_chunk.to_protobuf());
+        // execute on the compute node
+        let res = execute(fast_insert_request, compute_client).await?;
+
+        if res.status == fast_insert_response::Status::Succeeded as i32 {
+            Ok(())
+        } else {
+            Err(err(
+                anyhow!("Failed to fast insert: {}", res.error_message),
+                StatusCode::INTERNAL_SERVER_ERROR,
+            ))
+        }
+    }
 
-        let insert_stmt = Statement::Insert {
-            table_name: ObjectName::from(vec![table.as_str().into()]),
-            columns: vec![],
-            source: Box::new(Query {
-                with: None,
-                body: SetExpr::Values(Values(vec![vec![Expr::Value(Value::SingleQuotedString(
-                    payload,
-                ))]])),
-                order_by: vec![],
-                limit: None,
-                offset: None,
-                fetch: None,
-            }),
-            returning: vec![],
+    async fn acquire_table_info(
+        request_id: u32,
+        database: &String,
+        schema: &String,
+        table: &String,
+    ) -> Result<FastInsertContext> {
+        let session_mgr = SESSION_MANAGER
+            .get()
+            .expect("session manager has been initialized");
+
+        let frontend_env = session_mgr.env();
+
+        let search_path = SearchPath::default();
+        let schema_path = SchemaPath::new(Some(schema.as_str()), &search_path, USER);
+
+        let (webhook_source_info, table_id, version_id) = {
+            let reader = frontend_env.catalog_reader().read_guard();
+            let (table_catalog, _schema) = reader
+                .get_any_table_by_name(database.as_str(), schema_path, table)
+                .map_err(|e| err(e, StatusCode::NOT_FOUND))?;
+
+            let webhook_source_info = table_catalog
+                .webhook_info
+                .as_ref()
+                .ok_or_else(|| {
+                    err(
+                        anyhow!("Table `{}` is not with webhook source", table),
+                        StatusCode::FORBIDDEN,
+                    )
+                })?
+                .clone();
+            (
+                webhook_source_info,
+                table_catalog.id(),
+                table_catalog.version_id().expect("table must be versioned"),
+            )
         };
 
-        let _rsp = handle(session, insert_stmt, Arc::from(""), vec![])
+        let fast_insert_request = FastInsertRequest {
+            table_id: table_id.table_id,
+            table_version_id: version_id,
+            column_indices: vec![0],
+            // leave the data_chunk empty for now
+            data_chunk: None,
+            row_id_index: Some(1),
```

Comment:
I am a little bit confused by the `row_id_index`. What does the `row_id_index: Some(1)` mean here?

From the code, I think it means the position to insert a row_id column

While the comments said:

```
  // An optional field and will be `None` for tables without user-defined pk.
  // The `BatchInsertExecutor` should add a column with NULL value which will
  // be filled in streaming.
```

which is correct?

Reply:
Updated. It should use the `row_id_index` in `TableCatalog`. Also added test for it.